test/ConfigManagerTest.cpp

```diff
@@ -319,4 +321,505 @@ TEST(ConfigManagerTest, PrintConfigurationDocExistence) {
   ASSERT_NO_THROW(config.printConfigurationDoc(false));
   ASSERT_NO_THROW(config.printConfigurationDoc(true));
 }
+
+/*
+@brief Quick check, if `parseConfig` works with the registered validators.
+
+@param jsonWithValidValues When parsing this, no exceptions should be thrown.
+@param jsonWithNonValidValues When parsing this, an exception should be thrown.
+@param containedInExpectedErrorMessage What kind of text should be contained in
+the exception thrown for `jsonWithNonValidValues`. Validators have custom
+exception messages, so they should be identifiable.
+*/
+void checkValidator(ConfigManager& manager,
+                    const nlohmann::json& jsonWithValidValues,
+                    const nlohmann::json& jsonWithNonValidValues,
+                    std::string_view containedInExpectedErrorMessage) {
+  ASSERT_NO_THROW(manager.parseConfig(jsonWithValidValues));
+  AD_EXPECT_THROW_WITH_MESSAGE(
+      manager.parseConfig(jsonWithNonValidValues),
+      ::testing::ContainsRegex(containedInExpectedErrorMessage));
+}
+
+TEST(ConfigManagerTest, ConfigOptionWithValidator) {
+  ConfigManager managerWithoutSubManager{};
+  int notUsedInt;
+  managerWithoutSubManager.addOption("h", "", &notUsedInt)
+      .getConfigOption()
+      .addValidator([](const int& n) { return 0 <= n && n <= 24; },
+                    "Validator of h");
+  checkValidator(managerWithoutSubManager,
+                 nlohmann::json::parse(R"--({"h" : 10})--"),
+                 nlohmann::json::parse(R"--({"h" : 100})--"), "Validator of h");
+}
```

Comment:
Maybe you could add a few more test cases (also with multiple options that interact) to this "human-readable" test case, because the exhaustive templated tests below are hard to read for users who look for guidance. (just two or three)

Reply:
Like this [db601da](https://github.com/ad-freiburg/qlever/pull/1058/commits/db601da46d100ca3bed34b199c064b50f6c3fe8b)?